statsmodels/distributions/empirical_distribution.py

```diff
@@ -144,6 +144,52 @@ def __init__(self, x, side='right'):
         # which would have raised a NameError if hit, so would need to be
         # fixed.  See GH#5701.
 
+class WeightedECDF(StepFunction):
+    """
+    Return the Empirical Weighted CDF of an array as a step function.
+
+    Parameters
+    ----------
+    x : array_like
+        Observations
+    w : array_like
+        Weights of the observations
+    side : {'left', 'right'}, optional
+        Default is 'right'. Defines the shape of the intervals constituting the
+        steps. 'right' correspond to [a, b) intervals and 'left' to (a, b].
+
+    Returns
+    -------
+    Weighted ECDF as a step function.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from statsmodels.distributions.empirical_distribution import WeightedECDF
+    >>>
+    >>> ewcdf = WeightedECDF([3, 3, 1, 4], [1, 1, 1, 1])  # For equal weights ewcdf result is same as ecdf (see above)
+    >>> ewcdf([3, 55, 0.5, 1.5])
+    array([0.75, 1.  , 0.  , 0.25])
+    >>>
+    >>> ewcdf = WeightedECDF([3, 3, 1, 4], [0.25, 1, 2.5, 5])
+    >>>
+    >>> ewcdf([3, 55, 0.5, 1.5])
+    array([0.42857143, 1., 0. , 0.28571429])
+    """
+    def __init__(self, x, w, side='right'):
```

Comment:
use `weights` instead of `w` as keyword

(I guess those are frequency or case weights)